app/src/main/java/com/example/unlibrary/library/LibraryRepository.java

```diff
@@ -418,4 +425,126 @@ public void declineRequester(String requestedUID, String bookRequestedID, OnSucc
+                                String id = r.getId();
+                                DocumentReference requestDocument = mDb.collection(REQUESTS_COLLECTION).document(id);
+                                DocumentSnapshot snapshot = transaction.get(requestDocument);
+
+                                if (snapshot.get(REQUESTER).equals(requestedUID)) {
+                                    acceptedRequestDocument = requestDocument;
+
+                                } else {
+                                    declinedRequestDocuments.add(requestDocument);
+                                }
+                            }
+
+                            if (acceptedRequestDocument != null) {
+                                transaction.update(acceptedRequestDocument, LOCATION, new GeoPoint(handoffLocation.latitude, handoffLocation.longitude));
+                                transaction.update(acceptedRequestDocument, STATE, Request.State.ACCEPTED.toString());
+                            }
+
+                            for (DocumentReference doc : declinedRequestDocuments) {
+                                transaction.update(doc, STATE, ARCHIVED.toString());
+                            }
+
+
+                            final DocumentReference bookDocument = mDb.collection(BOOKS_COLLECTION).document(bookRequestedID);
+                            transaction.update(bookDocument, STATUS, Status.ACCEPTED.toString());
+
+                            // Success
+                            return null;
+                        }).addOnSuccessListener(onSuccessListener).addOnFailureListener(onFailureListener);
+                    } else {
+                        Log.e(TAG, "Error in fetching requests", requestTask.getException());
+                    }
+                });
+    }
+
+    /**
+     * Updates the handoff location for the book
+     *
+     * @param requestedUID      accepted requester user ID
+     * @param bookRequestedID   book ID request is associated with
+     * @param handoffLocation   LatLng of handoff location
+     * @param onSuccessListener code to call on success
+     * @param onFailureListener code to call on failure
+     */
+    public void updateHandoffLocation(String requestedUID, String bookRequestedID, LatLng handoffLocation, OnSuccessListener onSuccessListener, OnFailureListener onFailureListener) {
+        mDb.collection(REQUESTS_COLLECTION)
+                .whereEqualTo(REQUESTER, requestedUID)
+                .whereEqualTo(BOOK, bookRequestedID)
+                .whereNotEqualTo(STATE, ARCHIVED)
+                .get()
+                .addOnCompleteListener(task -> {
+                    if (task.isSuccessful()) {
+                        List<Request> requests = task.getResult().toObjects(Request.class);
+                        if (requests.size() != 1) {
+                            onFailureListener.onFailure(new Exception("Error updating handoff location"));
```

Comment:
Nit: I would make the log string more descriptive so we can differentiate it from other types of error that may occur during update.

Something like 
```suggestion
                            onFailureListener.onFailure(new Exception("Unexpected number of requests returned during update.  " + requests.size() + " requests found."));
```